### PR TITLE
chore: move try_read_attr function into table.lua (#2257)

### PR DESCRIPTION
**Pull request by starsz**
### What this PR does / why we need it:

fix https://github.com/apache/apisix/issues/2557

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: apisix/core/table.lua [hunk 1]
```diff
@@ -59,6 +59,22 @@ function _M.set(tab, ...)
 end
 
 
+function _M.try_read_attr(tab, ...)
```
Comment (spacewander) [inline apisix/core/table.lua:62]: Should also merge the `try_attr` in other plugin to this function.

Comment (starsz) [reply apisix/core/table.lua:62]: Sorry, I forgot.

File: apisix/plugins/hmac-auth.lua [hunk 2]
```diff
@@ -29,6 +28,7 @@ local core       = require("apisix.core")
 local hmac       = require("resty.hmac")
 local consumer   = require("apisix.consumer")
 local ngx_decode_base64 = ngx.decode_base64
+local try_read_attr = core.table.try_read_attr
```
Comment (membphis) [inline apisix/plugins/hmac-auth.lua:31]: we do not need to cache it, call `core.table.try_read_attr` is simpler.

Comment (membphis) [reply apisix/plugins/hmac-auth.lua:31]: please fix the similar points

Comment (starsz) [reply apisix/plugins/hmac-auth.lua:31]: OK. Let me fix those points.

File: apisix/plugins/hmac-auth.lua [hunk 3]
```diff
@@ -328,7 +312,7 @@ local function get_params(ctx)
     local date_key = DATE_KEY
     local signed_headers_key = SIGNED_HEADERS_KEY
 
-    if try_attr(local_conf, "plugin_attr", "hmac-auth") then
+    if core.table.try_read_attr(local_conf, "plugin_attr", "hmac-auth") then
         local attr = local_conf.plugin_attr["hmac-auth"]
```
Comment (spacewander) [inline apisix/plugins/hmac-auth.lua:316]: `try_read_attr` already gets what we need, we don't need to read it again.

File: apisix/plugins/log-rotate.lua [hunk 4]
```diff
@@ -152,25 +150,11 @@ local function scan_log_folder()
...
-end
-
-
 local function rotate()
     local local_conf = core.config.local_conf()
     local interval = INTERVAL
     local max_kept = MAX_KEPT
-    if try_attr(local_conf, "plugin_attr", "log-rotate") then
+    if core.table.try_read_attr(local_conf, "plugin_attr", "log-rotate") then
         local attr = local_conf.plugin_attr["log-rotate"]
```
Comment (spacewander) [inline apisix/plugins/log-rotate.lua:158]: Ditto

**Review by spacewander — CHANGES_REQUESTED** [review]

**Review by spacewander — APPROVED** [review]

File: apisix/plugins/hmac-auth.lua (hunk 3 above)
Comment (starsz) [reply apisix/plugins/hmac-auth.lua]: OK. I had fixed.

File: apisix/plugins/log-rotate.lua (hunk 4 above)
Comment (starsz) [reply apisix/plugins/log-rotate.lua]: OK. I had fixed.

**Review by membphis — APPROVED** [review]

**Pull request merged**